Gordon360/Services/RecIM/MatchService.cs

```diff
@@ -130,28 +145,30 @@ public MatchExtendedViewModel GetMatchByID(int matchID)
                             Username = mp.ParticipantUsername
                         }).AsEnumerable(),
                     
-                    SeriesID = m.SeriesID,
-                    // Team will eventually be handled by TeamService 
-                    Activity = _context.Activity
-                        .Where(a => a.ID == m.Series.ActivityID)
-                        .Select(a => new ActivityExtendedViewModel
+                    Series = _context.Series
+                        .Include(s => s.SeriesTeam)
+                            .ThenInclude(s => s.Team)
+                        .Where(s => s.ID == m.SeriesID)
+                        .Select(s => new SeriesExtendedViewModel
                         {
-                            ID = a.ID,
-                            Name = a.Name,
-                            Team = a.Team.Select(t => new TeamExtendedViewModel
-                            {
-                                ID = t.ID,
-                                Name = t.Name,
-                                Logo = t.Logo
-                            })
-                        })
-                        .FirstOrDefault(),
+                            ID = s.ID,
+                            Name = s.Name,
+                            Type = s.Type.Description,
+                            TeamStanding = s.SeriesTeam.Select(st => (TeamRecordViewModel)st)
+                        }).FirstOrDefault(),
+                    // Team will eventually be handled by TeamService 
+                    Activity = new ActivityExtendedViewModel
+                    {
+                        ID = m.Series.ActivityID,
+                        Name = m.Series.Activity.Name
+                    },
                     Team = m.MatchTeam.Select(mt => new TeamExtendedViewModel
                     {
                         ID = mt.TeamID,
                         Name = mt.Team.Name,
                         Status = mt.Status.Description,
                         Participant = mt.Team.ParticipantTeam
+                            .Where(pt => !new int[] {0,1,2}.Contains(pt.RoleTypeID)) //roletype is either deleted, invalid, invited to join
```

Comment:
For `{0, 1, 2}`, we should create a const outside the function to maintain the consistency if this exists more than one place. Also easier for the future developers to find where this forbidden roletype group is.

Reply:
This specific set won't exist elsewhere, but your point still doest make sense.